api/tests/mocha/services/settings.spec.js

```diff
@@ -0,0 +1,53 @@
+const sinon = require('sinon').sandbox.create();
+require('chai').should();
+
+const service = require('../../../src/services/settings'),
+      db = require('../../../src/db');
+
+let settings,
+    replace,
+    overwrite;
+
+describe('settings service', () => {
+  beforeEach(function() {
+    settings = { a: 'a', permissions: { b: 'b'} };
+    sinon.stub(db.medic, 'get').resolves({ settings });
+  });
+
+  afterEach(function() {
+    sinon.restore();
+  });
+
+  describe('update', () => {
+    it('does not overwrite if replace is set and overwrite is set', () => {
+      const update = sinon.stub(db.medic, 'put');
+      let newSettings = Object.assign({}, settings);
+      delete newSettings['c'];
```

Comment:
As far as I can tell there is no `newSettings['c']` so this deletion does nothing.